src/main/java/teammates/ui/webapi/action/GetCoursesAction.java

```diff
@@ -1,12 +1,18 @@
 package teammates.ui.webapi.action;
 
-import java.util.Comparator;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
+import teammates.common.datatransfer.attributes.AccountAttributes;
 import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.ui.webapi.output.CoursesData;
 
 /**
- * Get a list of course for current user.
+ * Gets all courses (active, archived and soft-deleted courses) for the instructor.
```

Comment:
What about students? Is get `/courses` only for instructors?

Reply:
Context for this:
long story short STUDENT_COURSES (GET) is the student API for courses, so it was placed under GET /course. However since STUDENT_COURSES returns a list of courses it shouldn't go under there, so Xiao Pu moved it to /courses. That was yesterday, so you might have not seen the updates.

With this change you will have to implement student functionality for /courses as well.